Sources/DangerSwiftKantoku/Kantoku.swift

```diff
@@ -77,19 +78,43 @@ extension Kantoku {
     
 }
 
+public typealias File = String
+extension File {
+    var finalFileName: String {
+        return self.components(separatedBy: "/").last ?? ""
+
+    }
+}
+
 extension Kantoku {
     
-    private func postIssuesIfNeeded(from resultFile: XCResultFile, configuration: XCResultParsingConfiguration) {
+    public enum WariningFilter {
+        case all
+        case modifiedAndCreatedFiles
+        case files([File])
+    }
+    
+    private func postIssuesIfNeeded(from resultFile: XCResultFile, configuration: XCResultParsingConfiguration, warnFor: WariningFilter = .all) {
         
         if configuration.needsIssues {
             
             guard let issues = resultFile.getInvocationRecord()?.issues else {
                 warn("Failed to get invocation record from \(resultFile.url.absoluteString)")
                 return
             }
-            
+
+            var targetFileNames: [String]? = nil
+            switch warnFor {
+            case .all : break
+            case .files(let files) :
+                targetFileNames = files.map{ $0.finalFileName }
+            case .modifiedAndCreatedFiles :
+                let allFilePaths = Danger().git.modifiedFiles + Danger().git.createdFiles
+                targetFileNames = allFilePaths.map { $0.finalFileName }
+            }
```

Comment:
[advice]
And here now we can filter files before calling `post` methods like this:

```swift
extension Kantoku {
    
    private func summaries<T: PostableIssueSummary>(of summaries: [T], filteredBy fileType: XCResultParsingConfiguration.ReportingFileType) -> [T] {
        
        let filteringPredicate: (XCResultParsingConfiguration.RelativeFilePath) -> Bool
        
        switch fileType {
        case .all:
            return summaries
            
        case .modifiedAndCreatedFiles:
            filteringPredicate = { (modifiedFiles + createdFiles).contains($0) }
            
        case .custome(predicate: let predicate):
            filteringPredicate = predicate
        }
        
        return summaries.filter { summary in
            guard let relativePath = summary.documentLocation?.relativePath(against: workingDirectoryPath) else {
                return false
            }
            return filteringPredicate(relativePath.filePath)
        }
        
    }

}
```

Now we can call post methods like this:

```diff
-                post(issues.warningSummaries, as: .warning)
+                let warningSummaries = summaries(of: issues.warningSummaries, filteredBy: configuration.reportingFileType)
+                post(warningSummaries, as: .warning)
```

Reply:
yup will do

Reply:
verified that the paths are relative to working directory
<img width="711" alt="Screenshot 2022-08-18 at 3 18 07 PM" src="https://user-images.githubusercontent.com/1806148/185366155-d796a747-f3f2-4117-8126-9476672d3ff0.png">